pkgs/applications/science/math/sage/sagelib.nix

```diff
@@ -87,83 +94,93 @@ buildPythonPackage rec {
 
   nativeBuildInputs = [
     iml
+    lisp-compiler
+    m4
     perl
-    jupyter-core
+    pip # needed to query installed packages
     pkg-config
     sage-setup
-    pip # needed to query installed packages
-    lisp-compiler
-    m4
   ];
 
   buildInputs = [
     gd
-    readline
     iml
     libpng
+    readline
   ];
 
   propagatedBuildInputs = [
-    cypari2
-    jinja2
-    numpy
-    pkgconfig
-    boost
+    # native dependencies
```

Comment:
This has nothing to do with cross compilation, right? If so maybe propagatedNativeBuildInputs is a better fit?

Reply:
That's a good point, I will check. Thanks!

Reply:
non-python dependencies should probably not be propagated at all.

Reply:
Technically, this isn't a regression from this PR, as the above changes just split previously-existing dependencies into two categories according to whether they're listed in sagelib's `requirements.txt` or not. Since most of those dependencies are binaries which will be executed at runtime by Sage, a proper split will require a bit of tweaking. I will do it in a follow-up PR.